src/app/components/todolist/todolist.component.ts

```diff
@@ -21,14 +21,7 @@ export class TodolistComponent implements OnInit {
     // starting state of the todo list
     this.todoTitle = '';
     this.idForTodo = 1;
```

Comment:
This is quite dirty, we have a global variable here that is being incremented every time we add something. It works, but not ideal.

I know you do not have a backend DB for it to be auto incremented, but what I would suggest in this scenario is to work out the next ID that needs to be set. So figure out the MAX Id in the current todos and then +1 that.

Reply:
Right i've had a go at a solution for this.  You'll see when i update the PR.  I first of all work out the maxid of the todo array.  Then if the maxid is >= the idForTodo we increment the idForTodo.